brainscore/benchmarks/nsd_bocini2023_algonauts.py

```diff
@@ -0,0 +1,172 @@
+  year={2022},
+  publisher={Nature Publishing Group US New York}
+}
+
+@article{gifford2023algonauts,
+  title={The algonauts project 2023 challenge: How the human brain makes sense of natural scenes},
+  author={Gifford, Alessandro T and Lahner, Benjamin and Saba-Sadiya, Sari and Vilas, Martina G and Lascelles, Alex and Oliva, Aude and Kay, Kendrick and Roig, Gemma and Cichy, Radoslaw M},
+  journal={arXiv preprint arXiv:2301.03198},
+  year={2023}
+}
+"""
+
+SUBJECTS = ["subject1", "subject2", "subject3", "subject4", "subject5", "subject6", "subject7", "subject8"]
+
+
+### BENCHMARK BASE #############################################################################################################
+
+class NSDNeuralBenchmark(BenchmarkBase):
+    def __init__(self, identifier, assemblies, similarity_metric, visual_degrees, number_of_trials, ceiler, **kwargs):
+        super(NSDNeuralBenchmark, self).__init__(identifier=identifier, **kwargs)
+        self._assemblies = assemblies
+        self._similarity_metric = similarity_metric
+        region = np.unique(self._assemblies[0]['region'])
+        assert len(region) == 1
+        self.region = region[0]
+        timebins = timebins_from_assembly(self._assemblies[0])
+        self.timebins = timebins
+        self._visual_degrees = visual_degrees
+        self._number_of_trials = number_of_trials
+        self.ceiling_func = ceiler
+
+    def __call__(self, candidate: BrainModel):
+        final_scores = []
+        for i, subject in tqdm(enumerate(SUBJECTS), desc='Computing raw score for all Subjects', total=len(SUBJECTS)):
+            print('start recording ANN responses')
+            candidate.start_recording(self.region, time_bins=self.timebins)
+            stimulus_set = place_on_screen(self._assemblies[i].stimulus_set, target_visual_degrees=candidate.visual_degrees(),
+                                    source_visual_degrees=self._visual_degrees)
+            source_assembly = candidate.look_at(stimulus_set, number_of_trials=self._number_of_trials)
+            print('source_assembly extracted')
+            if 'time_bin' in source_assembly.dims:
+                source_assembly = source_assembly.squeeze('time_bin')  # static case for these benchmarks
+            source_assembly = source_assembly.transpose('presentation', 'neuroid')
+            print('computing the raw score')
+            raw_score = self._similarity_metric(source_assembly, self._assemblies[i])
+            ceil = self.ceiling_func(self._assemblies[i])
+            
+            normalized_score = normalize_with_NC(raw_score, ceil)
+
+            final_scores = np.concatenate((final_scores, normalized_score))
+
+        score_center = np.mean(final_scores)
+        score_error = np.std(final_scores) / np.sqrt(len(final_scores))
+
+        score =  Score([score_center, score_error],
+                            coords={**{'aggregation': ['center', 'error']}},
+                            dims=('aggregation',) )
```

Comment:
For different subjects we have different number of stimuli. Hence the following code: 

subject_scores = Score.merge(*subject_scores)

returns the error:
 *** ValueError: arguments without labels along dimension 'dim_0' cannot be aligned because they have different dimension sizes: {1154, 1211} 

Any easy fix?

Reply:
shouldn't a subject score be a single number? the number of stimuli at that stage should not play a role